modules/core/src/test/java/org/apache/ignite/internal/processors/cache/GridCacheAbstractMetricsSelfTest.java

```diff
@@ -674,6 +674,59 @@ public void testRemoves() throws Exception {
         assertEquals(1L, cache.localMetrics().getCacheRemovals());
     }
 
+    /**
+     * Test {@link CacheMetrics#getSize()}, {@link CacheMetrics#getSizeLong()}, {@link CacheMetrics#getKeySize()} and
+     * {@link CacheMetrics#getKeySizeLong()}.
+     *
+     * @throws Exception If failed.
+     */
+    public void testSize() throws Exception {
+        IgniteCache<Integer, Integer> cache = grid(0).cache(DEFAULT_CACHE_NAME);
+
+        assertEquals(0, cache.metrics().getSize());
+        assertEquals(0L, cache.metrics().getSizeLong());
+
+        // Metrics works strange with the LOCAL cache mode.
```

Comment:
Why it works strangely? Please, describe it. And file a ticket.

Reply:
Make comment more clear.